Source/MQTTnet/Server/Internal/MqttClientSubscriptionsManager.cs

```diff
@@ -177,6 +178,13 @@ public async Task<SubscribeResult> Subscribe(MqttSubscribePacket subscribePacket
 
             var addedSubscriptions = new List<string>();
             var finalTopicFilters = new List<MqttTopicFilter>();
+            var atLeastOnceSubscriptionResults = new List<CreateSubscriptionResult>();
+
+            IList<MqttApplicationMessage> retainedApplicationMessages = null;
```

Comment:
The code does not compile on my machine because the variable _retainedApplicationMessages_ is already defined above...